packages/react-charting/src/components/GroupedVerticalBarChart/GroupedVerticalBarChart.base.tsx

```diff
@@ -392,33 +393,58 @@ export class GroupedVerticalBarChartBase extends React.Component<
       // use the following addend.
       const xPoint = xScale1(datasetKey) + (xScale1.bandwidth() - this._barWidth) / 2;
       const yPoint = Math.max(containerHeight! - this.margins.bottom! - yBarScale(pointData.data), 0);
+      let startColor = pointData.color ? pointData.color : getNextColor(index, this.props.theme?.isInverted);
+      let endColor = startColor;
+
+      if (this.props.enableGradient) {
+        startColor = pointData.gradient?.[0] || getNextGradient(index, 0, this.props.theme?.isInverted)[0];
+        endColor = pointData.gradient?.[1] || getNextGradient(index, 0, this.props.theme?.isInverted)[1];
+        pointData.color = startColor;
+      }
+
       // Not rendering data with 0.
       pointData.data &&
         singleGroup.push(
-          <rect
-            className={this._classNames.opacityChangeOnHover}
-            key={`${singleSet.indexNum}-${index}`}
-            height={Math.max(yBarScale(pointData.data), 0)}
-            width={this._barWidth}
-            x={xPoint}
-            y={yPoint}
-            data-is-focusable={
-              !this.props.hideTooltip && (this._legendHighlighted(pointData.legend) || this._noLegendHighlighted())
-            }
-            opacity={this._getOpacity(pointData.legend)}
-            ref={(e: SVGRectElement | null) => {
-              this._refCallback(e!, pointData.legend, refIndexNumber);
-            }}
-            fill={pointData.color}
-            onMouseOver={this._onBarHover.bind(this, pointData, singleSet)}
-            onMouseMove={this._onBarHover.bind(this, pointData, singleSet)}
-            onMouseOut={this._onBarLeave}
-            onFocus={this._onBarFocus.bind(this, pointData, singleSet, refIndexNumber)}
-            onBlur={this._onBarLeave}
-            onClick={this.props.href ? this._redirectToUrl.bind(this, this.props.href!) : pointData.onClick}
-            aria-label={this._getAriaLabel(pointData, singleSet.xAxisPoint)}
-            role="img"
-          />,
+          <React.Fragment key={`${singleSet.indexNum}-${index}`}>
+            {this.props.enableGradient && (
+              <defs>
+                <linearGradient
+                  id={`gradient_${index}_${singleSet.indexNum}_${endColor}`}
```

Comment:
use getId function to generate the prefix for a new id.
We don't want ids to collide if there are 2 or more charts on the same page